### PR TITLE
Fix ContextualMenu getItemClassNames memoization bug due to API change

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

- Fixes a `ContextualMenuItem` memoization issue due to the API change in #6094 which passes an `Object` rather than an argument list of primitives to the memoized function `getItemClassNames`.
- Fixes a regression for partners due to an API change of a _private_ function `getItemClassNames` in #6094.

**The Changes:**

Due to the _prettier_ changes, I suggest following the commits as I have them listed below for readability.

1. (https://github.com/OfficeDev/office-ui-fabric-react/commit/df816dce8b41d280350c9d8895f14b1c8c69f4a3) Reverts the API change of `getItemClassNames` and introduces a wrapper function `getItemStyles` which complies with `getStyles` signature.
1. (https://github.com/OfficeDev/office-ui-fabric-react/commit/ed1f7773f2e278d88667233ed65060d9984db719) Replaces the usage of `getItemClassNames` as `getStyles` function with `getItemStyles`.

**Regarding Version Number**

This PR is marked as a `patch` version because the API surface being modified is not part of the public API surface. It is not a top-level export in OUIFR.

#### Focus areas to test

I've tested the following scenarios to ensure that they work.

- The unit test suite including the tests added as part of #6094 which cover both `ContextualMenu` and `ContextualMenuItem` styling (deprecated and not) and `getStyles` should succeed.
- One should be able to customize the styles of a `ContextualMenu` via `IContextualMenuProps#styles`.
- One should be able to customize individual styles of a `ContextualMenuItem` via its `itemProps: IContextualMenuItemProps`'s `styles` attribute.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts [hunk 1]
```diff
@@ -86,19 +86,21 @@ export const getItemStyles = (props: IContextualMenuItemStyleProps): IContextual
...
-    checked,
-    isAnchorLink,
-    knownIcon,
-    itemClassName,
-    dividerClassName,
-    iconClassName,
-    subMenuClassName,
-    primaryDisabled,
-    className
+  return <IContextualMenuItemStyles>(
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts]: Hmm this is incorrect, working with @JasonGore

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts [hunk 2]
```diff
@@ -66,21 +66,20 @@ const GlobalClassNames = {
 // TODO: Audit perf. impact of and potentially remove memoizeFunction.
 //       https://github.com/OfficeDev/office-ui-fabric-react/issues/5534
 export const getItemClassNames = memoizeFunction(
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts]: the original code had memoize too?

I would add jsdoc stating that this is package-internal method that has unfortunately been depended upon and is being kept in this form for backwards compat, but will be cleaned up by Fabric 7. 

Also add `@deprecated`.

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts]: Yea, it used `memoizeFunction` prior to #6094. Regarding the annotation, I'd prefer to use `@internal` in preparation for API Extractor. This function does all of the `getStyles` style-set merging so its not `@deprecated` per say. However it should not be relied on as a _public_ API. It is our internal mechanism for generating `ContextualMenuItem` classNames given style props.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItem.test.tsx [hunk 3]
```diff
@@ -109,6 +104,65 @@ describe('ContextMenuItemChildren', () => {
   });
 });
 
+describe('getItemClassNames', () => {
+  // This test exists to validate that getItemClassNames signature has not changed
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItem.test.tsx]: Can you add this instead to contextualMenu.deprecated.tests.ts? (We have this pattern throughout the repo of tests supporting deprecated props/methods)

**Pull request merged**